spark/src/test/java/org/apache/zeppelin/spark/PySparkInterpreterTest.java

```diff
@@ -120,4 +122,33 @@ public void testCompletion() {
       assertTrue(completions.size() > 0);
     }
   }
+
+  private class infinityPythonJob implements Runnable {
+    @Override
+    public void run() {
+      String code = "import time\nwhile True:\n  time.sleep(1)" ;
+      InterpreterResult ret = pySparkInterpreter.interpret(code, context);
+      assertNotNull(ret);
+      Pattern expectedMessage = Pattern.compile("KeyboardInterrupt");
+      Matcher m = expectedMessage.matcher(ret.message().toString());
+      assertTrue(m.find());
+    }
+  }
+
+  @Test
+  public void testCancelIntp() {
+    if (getSparkVersionNumber() > 11) {
+
+      assertEquals(InterpreterResult.Code.SUCCESS,
+        pySparkInterpreter.interpret("a = 1\n", context).code());
+
+      new Thread(new infinityPythonJob()).start();
+      try {
+        Thread.sleep(5000);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+      pySparkInterpreter.cancel(context);
```

Comment:
Don't we need to something that checks if thread is actually finished or not?
I think `thread.join()` is required here. otherwise test will not fail even if job is not cancelled.